src/components/Product-card/ProductCard.jsx

```diff
@@ -28,8 +29,19 @@ export const ProductCard=({product})=>{
                             Add to cart
                         </button>
                     ) }
-                    
-                    <button className="btn btn-secondary">Add to wishlist</button>
+                    {
+                        wishlistState.wishlist.find((item) => item._id === _id) ? (
+                                <button className="btn btn-secondary"
+                                onClick={()=>wishlistDispatch({type:"RemoveFromWishlist",payload:_id})}>
```

Comment:
could some animation be done for on hover of just the button? 
also the buttons section and the top image /title part could use a bit of a gap in my opinion.